rust/src/doc.rs

```diff
@@ -500,6 +500,14 @@ impl Doc {
         doc.save()
     }
 
+    pub fn save_with_options(&self, message: String, time: i64) -> Vec<u8> {
```

Comment:
What's the reasoning behind combining `save` and `commit`? It feels to me like a nicer API would be to expose a `commit_with` method which just calls `doc.commit_with(..)`.